apps/frontend/src/views/Checklist.vue

```diff
@@ -0,0 +1,313 @@
+  }
+
+  /**
+   * The current search terms, as modeled by the search bar
+   */
+  get searchTerm(): string {
+    return SearchModule.searchTerm;
+  }
+
+  set searchTerm(term: string) {
+    SearchModule.updateSearch(term);
+  }
+
+  /**
+   * The currently selected file, if one exists.
+   * Controlled by router.
+   */
+  get fileFilter(): FileID[] {
+    return FilteredDataModule.selectedChecklistIds;
+  }
+
+  getChecklist(fileID: FileID[]) {
+    return InspecDataModule.allChecklistFiles.find(
+      (f) => f.uniqueId === fileID[0]
+    );
+  }
+
+  /**
+   * The filter for charts. Contains all of our filter stuff
+   */
+  get allFilter(): ChecklistFilter {
+    return {
+      status: SearchModule.inFileSearchTerms.statusFilter,
+      severity: SearchModule.inFileSearchTerms.severityFilter,
+      fromFile: this.fileFilter,
+      ruleidSearchTerms: SearchModule.inFileSearchTerms.ruleid,
+      vulidSearchTerms: SearchModule.inFileSearchTerms.vulid,
+      stigidSearchTerms: SearchModule.inFileSearchTerms.stigid,
+      classificationSearchTerms: SearchModule.inFileSearchTerms.classification,
+      groupNameSearchTerms: SearchModule.inFileSearchTerms.groupName,
+      cciSearchTerms: SearchModule.inFileSearchTerms.cci,
+      titleSearchTerms: SearchModule.inFileSearchTerms.title,
+      descriptionSearchTerms: SearchModule.inFileSearchTerms.description,
+      nistIdFilter: SearchModule.inFileSearchTerms.NISTIdFilter,
+      codeSearchTerms: SearchModule.inFileSearchTerms.code,
+      omit_overlayed_controls: true,
+      control_id: this.controlSelection || undefined,
+      iaControlsSearchTerms: SearchModule.inFileSearchTerms.iaControls,
+      keywordsSearchTerms: SearchModule.inFileSearchTerms.keywords
+    };
+  }
+
+  /**
+   * Clear all filters
+   */
+  clear(clearSearchBar = false) {
+    SearchModule.clear();
+    this.filterSnackbar = false;
+    this.controlSelection = null;
+    if (clearSearchBar) {
+      this.searchTerm = '';
+    }
+  }
+
+  /**
+   * Returns true if we can currently clear.
+   * Essentially, just controls whether the button is available
+   */
+  get canClear(): boolean {
+    // Return if any params not null/empty
+    let result: boolean;
+    if (
+      SearchModule.inFileSearchTerms.severityFilter.length !== 0 ||
+      SearchModule.inFileSearchTerms.statusFilter.length !== 0 ||
+      SearchModule.inFileSearchTerms.controlId.length !== 0 ||
+      SearchModule.inFileSearchTerms.code.length !== 0 ||
+      this.searchTerm
+    ) {
+      result = true;
+    } else {
+      result = false;
+    }
+
+    // Logic to check: are any files actually visible?
+    if (this.rules.length === 0) {
+      this.filterSnackbar = true;
+    } else {
+      this.filterSnackbar = false;
+    }
+
+    // Finally, return our result
+    return result;
+  }
+
+  get rules() {
+    const rulesList: ChecklistVuln[] = [];
+    this.getChecklist(this.fileFilter)
```

Comment:
feels like this is part of the holdover from the results.vue copy over.  there's only ever one checklist file selected.  get rid of the array nonsense.